neptune/internal/storage/datastream.py

```diff
@@ -65,6 +65,8 @@ def generate(self) -> Generator[FileChunk, Any, None]:
         while last_offset < self._total_size:
             chunk = self._fobj.read(chunk_size)
             if chunk:
+                if len(chunk) != min(chunk_size, self._total_size - last_offset):
```

Comment:
Do you think we can depend on mtime instead?

Reply:
I think we could -- when I think about it, it probably better illustrates "changed during upload", even if it sometimes restarts unnecessarily.
Will do that.